### PR TITLE
Sprint 45.5

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 45.5

- Development Team: Allow production domain configuration (Groom w/ 18F) ustaxcourt/ef-cms#7190
- Ability to easily reference case details while generating an order and QC ustaxcourt/ef-cms#7084
- Revisit flow: Proposed/Signed Stipulated Decision ustaxcourt/ef-cms#7205

Before deploying, please read: https://github.com/flexion/ef-cms/issues/5666#issuecomment-667339500

### CODE REVIEW

File: get-zone-name.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+elif [[ $BRANCH == 'test' ]] ; then
+  echo "${ZONE_NAME_TEST}"
+elif [[ $BRANCH == 'migration' ]] ; then
+  echo "${ZONE_NAME_MIG}"
+elif [[ $BRANCH == 'staging' ]] ; then
+  echo "${ZONE_NAME_STG}"
+elif [[ $BRANCH == 'master' ]] ; then
+  echo "${ZONE_NAME_PROD}"
+elif [[ $BRANCH == 'dawson' ]] ; then
+  echo "${ZONE_NAME_DAWSON}"
```
Comment (mmarcotte) [inline get-zone-name.sh]: Pretty sure for us this needs to be 
```
elif [[ $BRANCH == 'prod' ]] ; then
  echo "${ZONE_NAME_PROD}"
```
so, `s/dawson/prod`

**Comment by julialeague** [comment]
- [x] begin by deleting any existing copy of the Git repository, running `docker system prune` to remove any cached artifacts from any prior code review
- [x] clone a fresh copy of the Git repository (`git clone https://github.com/ustaxcourt/ef-cms.git`), fetch the pull request and name it for the sprint (e.g., `git fetch origin pull/333/head:sprint-45.5`), and then switch to that branch (e.g. `git checkout sprint-45.5`)
- [x] review the pull request itself, to get oriented
	- [x] read the description of the pull request, which should summarize the changes made
	- [x] read through every task on the Scrum board that's encompassed by this pull request
	- [x] read the description of the commits that comprise the pull request
- [x] stand up the site locally, with `./docker-run.sh`
	- [x] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [x] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [x] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [x] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [x] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [x] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [x] review for accessibility
		- [x] use an automated audit tool, such as Chrome Audit or aXe
		- [x] navigate site only with the keyboard
		- [x] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [x] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [x] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [x] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [x] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [x] review all tests
	- [x] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [x] methodically review all new tests for correctness, quality of naming
- [x] determine what code isn’t tested, review that rigorously
- [x] review documentation to ensure that it matches changes
- [x] provide comments on the pull request on GitHub, as necessary
	- [x] for comments that are specific to a particular line of code, comment on those specific lines
	- [x] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [x] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by adunkman** [comment]
While doing local testing, uncovered an issue where it appears IDs are not being set at `reports/case-deadlines` (notice the ID attributes are not assigned a value): 

> [DOM] Found 2 elements with non-unique id #: (More info: https://goo.gl/9p2vKq) 
> 
> ```html
><input aria-describedby=​"deadlineStart-date-start-label deadlineStart-date-start-hint" class=​"usa-input usa-sr-only usa-date-picker__internal-input" id name=​"deadlineStart-date-start" type=​"text" aria-hidden=​"true" tabindex=​"-1">​ 
><input aria-describedby=​"deadlineEnd-date-start-label deadlineEnd-date-start-hint" class=​"usa-input usa-sr-only usa-date-picker__internal-input" id name=​"deadlineEnd-date-start" type=​"text" aria-hidden=​"true" tabindex=​"-1">​
> ```

**Comment by adunkman** [comment]
Also found when local testing — the calendar on left date input at **Advanced Search > Order** doesn’t work in Firefox 79 — the button doesn’t respond to click events. 

Screen capture showing the issue described

File: web-api/domain-s3-backup.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -e
```
Comment (julialeague) [inline web-api/domain-s3-backup.sh:1]: @ericsorenson Could we get a description of the purpose of this script here?

File: web-api/domain-s3-restore.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash -e
```
Comment (julialeague) [inline web-api/domain-s3-restore.sh:1]: @ericsorenson Could we get a description for the purpose of this script?

File: web-api/src/documents/virusScanPdfLambda.js [hunk 4]
```diff
@@ -9,13 +9,10 @@ const { genericHandler } = require('../genericHandler');
...
   genericHandler(
     event,
-    async ({ applicationContext }) => {
-      const { documentId } = event.pathParameters || event.path || {};
-
-      return await applicationContext.getUseCases().virusScanPdfInteractor({
-        applicationContext,
-        documentId,
-      });
+    async () => {
```
Comment (julialeague) [inline web-api/src/documents/virusScanPdfLambda.js:12]: Is there a story logged to reactivate this later?

File: web-api/terraform/template/api.tf [hunk 5]
```diff
@@ -24,26 +43,28 @@ resource "aws_lambda_function" "api_lambda" {
...
 
-resource "aws_lambda_function" "api_clamav_lambda" {
-  filename      = data.archive_file.zip_api.output_path
-  function_name = "api_clamav_${var.environment}"
-  role          = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/lambda_role_${var.environment}"
-  handler       = "api.handler"
-  source_code_hash = data.archive_file.zip_api.output_base64sha256
-  timeout = "29"
-  memory_size = "3008"
+# TODO: The virus definitions have gotten too large to live on Lambda... find alternative solution to get
```
Comment (julialeague) [inline web-api/terraform/template/api.tf:46]: Is there a user story for this work?

File: web-api/src/documents/virusScanPdfLambda.js (hunk 4 above)
Comment (ericsorenson) [reply web-api/src/documents/virusScanPdfLambda.js:12]: Yes: https://app.zenhub.com/workspaces/flexionef-cms-5bbe4bed4b5806bc2bec65d3/issues/flexion/ef-cms/6053

File: web-api/terraform/template/api.tf (hunk 5 above)
Comment (ericsorenson) [reply web-api/terraform/template/api.tf:46]: Yes: https://app.zenhub.com/workspaces/flexionef-cms-5bbe4bed4b5806bc2bec65d3/issues/flexion/ef-cms/6053

**Comment by ericsorenson** [comment]
@adunkman date pickers were being worked on during the last 2 weeks. They're working now, and will be included in the Sprint 46 PR.

File: get-zone-name.sh (hunk 1 above)
Comment (ericsorenson) [reply get-zone-name.sh]: Updated!

File: web-api/domain-s3-backup.sh (hunk 2 above)
Comment (ericsorenson) [reply web-api/domain-s3-backup.sh:1]: It's a one time script for this PR, documented in https://github.com/flexion/ef-cms/issues/5666#issuecomment-667339500

File: web-api/domain-s3-restore.sh (hunk 3 above)
Comment (ericsorenson) [reply web-api/domain-s3-restore.sh:1]: It's a one time script for this PR, documented in https://github.com/flexion/ef-cms/issues/5666#issuecomment-667339500

File: web-client/src/presenter/actions/takePathForDocumentPreviewAction.test.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      },
+      props: {
+        fileFromBrowserMemory: {},
+      },
+    });
+
+    expect(presenter.providers.path.pdfInMemory).toHaveBeenCalled();
+  });
+
+  it('should return path.pdfInS3 when props.documentInS3 is defined', async () => {
```
Comment (julialeague) [inline web-client/src/presenter/actions/takePathForDocumentPreviewAction.test.js]: @ericsorenson Should `path.pdfInS3` be `path.documentsInS3`?

Comment (ericsorenson) [reply web-client/src/presenter/actions/takePathForDocumentPreviewAction.test.js]: Yes!

Comment (ericsorenson) [reply web-client/src/presenter/actions/takePathForDocumentPreviewAction.test.js]: Updated!

**Review by julialeague — APPROVED** [review]
@mmarcotte, I recommend you accept this pull request into the staging environment. Thanks!

**Comment by adunkman** [comment]
> @adunkman date pickers were being worked on during the last 2 weeks. They're working now, and will be included in the Sprint 46 PR.

Yay! @ericsorenson, can you confirm the missing ID issue at `reports/case-deadlines` is also resolved? (I think your comment meant both, but wanted to make sure it didn’t get lost).

**Comment by adunkman** [comment]
Confirmed with @ericsorenson vocally that the above issue is also resolved. I’m all 🚀 !

**Review by adunkman — APPROVED** [review]

**Comment by mmarcotte** [comment]
Thanks @adunkman , @julialeague , @ericsorenson , and everyone else!

**Pull request merged**